### PR TITLE
Fix allocating guid

### LINKED ISSUE #58: Networks can allocated GUID out of pool range
If the range is `02:00:00:00:00:00:00:00` - `02:00:00:00:00:00:00:FF`, Networks can allocated  GUID like `02:00:00:00:00:FF:00:00`, which is out of pool range.

### PR DESCRIPTION
**Pull request by Mmduh-483**
fixes #58

### CODE REVIEW

**Comment by Mmduh-483** [comment]
> Please add the Fixes #

Already added


> Also unit test are failing

Fixed

File: pkg/guid/guid.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package guid
+
+import (
+	"fmt"
+	"net"
+)
+
+// GUID address is an net.HardwareAddr encapsulation
+type GUID net.HardwareAddr
```
Comment (adrianchiris) [inline pkg/guid/guid.go:9]: perhaps type it as : InfinibandGUID ?
the term GUID is very general.

Comment (adrianchiris) [inline pkg/guid/guid.go:9]: i see alot of conversions to Uint64() in guid pool,

perhaps the type should  be `uint64` and the package would have a `func FromHardwareAddress(address net.HardwareAddr) GUID` ?

also add `GUID.HardwareAddress()` method for the other way around to be used in the daemon

File: pkg/guid/guid_pool.go [hunk 2]
```diff
@@ -169,55 +148,60 @@ func (p *guidPool) ReleaseGUID(guid string) error {
...
 	return nil
 }
 
-func (p *guidPool) getNextGUID(currentGUID net.HardwareAddr) net.HardwareAddr {
+func isValidRange(rangeStart, rangeEnd GUID) bool {
+	return rangeStart.Uint64() <= rangeEnd.Uint64() && rangeStart.Uint64() != 0 && rangeEnd.Uint64() != 0xFFFFFFFFFFFFFFFF
+}
+
+// nextGUID return next guid
+func nextGUID(guid GUID) GUID {
```
Comment (adrianchiris) [inline pkg/guid/guid_pool.go:181]: if GUID was a uint64 then you would return : guid + 1 (see my above suggestion)

**Comment by Mmduh-483** [comment]
> the second commit is related to PR#59 i dont understand why its here.

This PR request #59 , It is just the second commit should be reviewed

**Comment by moshe010** [comment]
@Mmduh-483 , maybe next time indicated that other PR need to be review before this one

**Comment by adrianchiris** [comment]
IMO this could have been merged without the dependency

**Comment by moshe010** [comment]
> IMO this could have been merged without the dependency

He is using the uint64 change to solve i, but anyway not important for now

**Pull request merged**